p03-form-validations/src/Form/Form.component.js

```diff
@@ -1,20 +1,85 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { Card } from '../Shared/components/Card'
 import { Button, Input } from '../Shared/components'
-import { fieldsList } from './Form.services'
+import { fieldsList, handleSubmitForm, isFieldValid } from './Form.services'
 
 import "./Form.styles.css"
 
+const formData={
+  name:'',
+  lastname:'',
+  birthdate: '',
+  country:'',
+  password:'',
+
+}
 const Form = () => {
+  const [ formState, setFormState ] = useState(formData);
+  const [hasValue, setHasValue]= useState(true);
+  const [fieldsForm,setFieldForm ]=useState(fieldsList);
+  useEffect(() => {
+      if(fieldsList.length  === fieldsForm.filter((field)=>field.hasvalue).length) 
+       {   
+        setHasValue(false);
+      }else{
+        setHasValue(true);
+      }
+
+  }, [formState])
   
+  const onInputChange = ({ target }) => {
+    const { name, value,type } = target;
+      setFormState({
+        ...formState,
+        [name]: value
+      });
+
+ 
+
+  const newList= fieldsForm.map((field)=>{
+          if (field.name===name && isFieldValid(value)){
+            const updateField={
+              ...field,
+              hasvalue:true,
+            }
+            return updateField
+          }
+
+          return field
+
+    })
+
+    setFieldForm(newList);
+       
+  }
+
+  const handleForm=()=>{
+    setFormState(formData);
+    setFieldForm(fieldsList);
+    setHasValue(true);
+    handleSubmitForm();
+    
+
+  }
+
+  const handleClear=()=>{
+   setFormState(formData)
+  }
+
   return (
     <Card width={"50%"}>
-      <div className="form">
-        <Input label="this is label" />
+      {
+       fieldsForm.map(({name,label,type})=>(
+      
+        <div className="form" key={name}>
+        <Input label={label} type={type} name={name} value={formState[name]} ChangeValue={onInputChange} />
       </div>
+
+       )) 
+      }
       <div className="form-actions">
-        <Button title="Clear"/>
-        <Button title="Subscribe"/>
+        <Button title="Clear" style={{color: "darkorange",  borderColor:'yellow'}} onclickButton={handleClear} />
```

Comment:
usually there is not a good practice to write the css inline on the element, this is because it could be harder to maintain in the future. What you can do is pass a className to the element and write the css on a separated file.